client_http.hpp

```diff
@@ -12,15 +11,40 @@
 #include <mutex>
 #include <type_traits>
 
+#ifdef USE_BOOST_ASIO
+#include <boost/asio.hpp>
+#define ASIO_NS boost::asio
```

Comment:
Why `#define ASIO_NS boost::asio`?  Why not use `using ASIO_NS = boost::asio;` instead?